hierarchy/hierarchy_test.go

```diff
@@ -185,90 +185,138 @@ func getMockHierarchyAPI(expectRequest http.Request, mockedHTTPResponse MockedHT
 func TestClient_GetRoot(t *testing.T) {
 	instanceID := "foo"
 	name := "bar"
-	model := `{"label":"","has_data":true}`
-	Convey("When bad request is returned", t, func() {
+
+	Convey("Given a bad request API response, then the expected error is returned", t, func() {
 		ts, mockedAPI := getMockHierarchyAPI(http.Request{Method: "GET"}, MockedHTTPResponse{StatusCode: 400, Body: ""})
 		_, err := mockedAPI.GetRoot(ctx, instanceID, name)
-		So(err, ShouldNotBeNil)
+		So(err, ShouldResemble, &ErrInvalidHierarchyAPIResponse{
+			actualCode:   400,
+			expectedCode: 200,
+			uri:          "/hierarchies/foo/bar",
+		})
 		ts.Close()
 	})
 
-	Convey("When server error is returned", t, func() {
+	Convey("Given a server error API response, then the expected error is returned", t, func() {
 		ts, mockedAPI := getMockHierarchyAPI(http.Request{Method: "GET"}, MockedHTTPResponse{StatusCode: 500, Body: "qux"})
 		mockedAPI.hcCli.Client.SetMaxRetries(2)
 		_, err := mockedAPI.GetRoot(ctx, instanceID, name)
-		So(err, ShouldNotBeNil)
+		So(err, ShouldResemble, &ErrInvalidHierarchyAPIResponse{
```

Comment:
Could it be worth extracting this into a helper function that builds the expected response for us? In case the expected response ever changes in the future (e.g. additional field added). We can then just pass the new argument instead

Reply:
There is a method to create the error struct, maybe we can use it:
```
func NewErrInvalidHierarchyAPIResponse(expectedCode, actualCode int, uri string) error {
```

Reply:
I used this method to generate the expected responses, but also added another test to test this method itself.